oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/FileStore.java

```diff
@@ -146,7 +140,7 @@ private static GarbageCollectionStrategy newGarbageCollectionStrategy() {
 
         this.segmentWriter = defaultSegmentWriterBuilder("sys")
                 .withGeneration(() -> getGcGeneration().nonGC())
-                .withWriterPool()
+                .withWriterPool(SegmentBufferWriterPool.PoolType.GLOBAL)
```

Comment:
Given my other comment about keeping withWriterPool(), I suggest reverting this change.